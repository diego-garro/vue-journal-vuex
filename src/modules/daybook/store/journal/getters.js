// export const myGetter = (state) => {
//   return state.algo
// }

export const getEntriesByTerm = (state) => (term = "") => {
  if (term.length === 0) return state.entries

  return state.entries.firlter((entry) =>
    entry.text.toLowerCase().includes(term.toLowerCase())
  )
}

export const getEntryById = (/*state*/) => {}
