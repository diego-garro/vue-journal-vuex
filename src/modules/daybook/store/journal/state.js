export default () => ({
  isLoading: true,
  entries: [
    {
      id: "1", // 12345875966325
      date: new Date().toDateString(), // sat, 23, julio
      text:
        "Ipsum veniam ipsum minim laborum fugiat officia laborum quis sunt id qui sunt Lorem ipsum.",
      picture: null,
    },
    {
      id: "2", // 12345875966325
      date: new Date().toDateString(), // sat, 23, julio
      text:
        "Veniam proident reprehenderit amet anim incididunt ullamco esse ipsum cupidatat commodo nisi ipsum officia aliquip.",
      picture: null,
    },
    {
      id: "3", // 12345875966325
      date: new Date().toDateString(), // sat, 23, julio
      text:
        "Eu et irure do consectetur ut deserunt fugiat aliqua ea ex esse incididunt nisi.",
      picture: null,
    },
  ],
})
