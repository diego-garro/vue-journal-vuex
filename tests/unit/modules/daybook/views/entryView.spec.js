import { shallowMount } from "@vue/test-utils"
import { createStore } from "vuex"

import journal from "@/modules/daybook/store/journal"
import { journalState } from "../../../mock-data/test-journal-state"
import EntryView from "@/modules/daybook/views/EntryView.vue"

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  })

describe("Pruebas en el EntryView component", () => {
  const store = createVuexStore(journalState)
  const mockRouter = {
    push: jest.fn(),
  }

  let wrapper

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallowMount(EntryView, {
      props: {
        id: "-Mo5dj4lF0GaZ4kCMAp0",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    })
  })

  test("debe de sacar al usuario porque el id no existe", () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: "este id no existe en el STORE",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    })

    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" })
  })

  test("debe de mostrar la entrada correctamente", () => {
    expect(wrapper.html()).toMatchSnapshot()
    expect(mockRouter.push).not.toHaveBeenCalledWith({ name: "no-entry" })
  })
})
