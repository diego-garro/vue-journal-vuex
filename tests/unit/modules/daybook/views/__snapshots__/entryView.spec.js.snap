// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView component debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">9</span><span class="mx-1 fs-3">Noviembre</span><span class="mx-2 fs-4 fw-light">2021, Martes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?"></textarea></div><img src="https://res.cloudinary.com/diebuntu/image/upload/v1636502343/l2nfoau7vy91raddkv5i.png" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
